modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/FunctionObjectValidationRuleEngineImpl.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.internal.validation.rule;
+
+import com.liferay.object.constants.ObjectValidationRuleConstants;
+import com.liferay.object.internal.configuration.FunctionObjectActionExecutorImplConfiguration;
+import com.liferay.object.validation.rule.ObjectValidationRuleEngine;
+import com.liferay.osgi.util.configuration.ConfigurationFactoryUtil;
+import com.liferay.portal.catapult.PortalCatapult;
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.util.Http;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Mateus Santana
+ */
+@Component(
+	configurationPid = "com.liferay.object.internal.configuration.FunctionObjectValidationRuleEngineImplConfiguration",
+	factory = "com.liferay.object.internal.configuration.FunctionObjectValidationRuleEngineImplConfiguration",
+	service = ObjectValidationRuleEngine.class
+)
+public class FunctionObjectValidationRuleEngineImpl
+	implements ObjectValidationRuleEngine {
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		_portalCatapult.launch(
+			_companyId, Http.Method.POST,
+			_functionObjectActionExecutorImplConfiguration.
+				oAuth2ApplicationExternalReferenceCode(),
+			_jsonFactory.createJSONObject(
+				_jsonFactory.looseSerialize(
+					notificationContext, "notificationRecipientSettings",
+					"termValues")),
+			_functionObjectActionExecutorImplConfiguration.resourcePath(),
+			notificationContext.getUserId());
+
+		return ObjectValidationRuleEngine.super.execute(inputObjects, script);
+	}
+
+	@Override
+	public String getName() {
+		return ObjectValidationRuleConstants.ENGINE_TYPE_FUNCTION;
```

Comment:
@mateussgn this is supposed to be a unique reference for each instance, 

this way you won't be able to differ the instances because the reference is always "function", [see this example](https://github.com/liferay/liferay-portal/blob/ab0cf42d75c9bdd549829d270c3e9019406af872/modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/type/FunctionNotificationType.java#L86)

Reply:
you should use `object.validation.rule` defined in the Factory

Reply:
I see! Thanks for the tip!

Reply:
yes